saritasa_s3_tools/client.py

```diff
@@ -27,32 +27,69 @@
 ]
 
 
-def get_boto3_s3_client(
+def get_boto3_session(
     access_key_getter: AccessKeyGetter,
-    s3_endpoint_url_getter: S3EndpointUrlGetter | None = None,
     region: RegionGetter | str = "",
-    max_pool_connections: int = 100,
-    signature_version: str | None = None,
-) -> mypy_boto3_s3.S3Client:
-    """Prepare boto3's s3 client for usage."""
-    endpoint_url = None
-    if s3_endpoint_url_getter:
-        endpoint_url = s3_endpoint_url_getter()
+) -> boto3.session.Session:
+    """Get AWS session."""
     if callable(region):
         region = region()
     credentials = access_key_getter()
-    return boto3.client(
-        service_name="s3",  # type: ignore
-        region_name=region,
+    return boto3.session.Session(
         aws_session_token=credentials.token or None,
         aws_access_key_id=credentials.access_key or None,
         aws_secret_access_key=credentials.secret_key or None,
+        region_name=region,
+    )
+
+
+def get_boto3_s3_client(
```

Comment:
`get_boto3_s3_resource` and `get_boto3_s3_client` are very similar.
Maybe prepare base private function with the same code? For example

```python
def _get_s3_tool(name: str):
    if access_key_getter:
        session = get_boto3_session(
            access_key_getter=access_key_getter,
            region=region,
        )
    if not session:
        raise ValueError("Please pass either session or access_key_getter")
    endpoint_url = None
    if s3_endpoint_url_getter:
        endpoint_url = s3_endpoint_url_getter()
    
    s3_tool = getattr(session, name, None)
    if not s3_tool:
        raise ValueError
    return s3_tool(
        service_name="s3",  # type: ignore
        endpoint_url=endpoint_url,
        config=config,
    )
```

Reply:
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Haha, ok, got it)